src/v/config/configuration.cc

```diff
@@ -1411,6 +1411,13 @@ configuration::configuration()
       "Enables rack-aware replica assignment",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
+  , storage_strict_data_init(
+      *this,
+      "storage_strict_data_init",
+      "Requires that an empty file named `.redpanda_data_dir` be present in "
+      "the data directory",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
```

Comment:
This property looks like something that should require a restart.

Reply:
I wasn't too sure about this. It technically doesn't need a restart to take effect. As soon as it's enable RP will look for a magic file in it's data directory on start. So is a restart required for a property that only effects RP starting?

Reply:
to me it looks like it should, but it won't hurt if there is no restart
maybe it's even better since it adds flexibility

Reply:
>So is a restart required for a property that only effects RP starting?

it's up to you. if needs_restart is no, then you can register a callback that that fires when the configuration value changes at runtime. another way to approach this is to ask the question: what would you want to happen if the option became active after redpanda started and the magic file were not present? would you want redpanda to immediately halt? if you just want it to take affect on the next boot, then i think you can change it to needs_restart = yes.

cc @jcsp to check my work here

Reply:
Honestly it's ambiguous.  I would lean toward saying needs_restart::yes, because it gives the user the right hint: if they modify the config, then look at a node's config status, it'll say the node requires restart, which is broadly true if the property is to have any effect (since it only does anything on restart).

The counterpoint would be that setting this property doesn't _require_ a restart because nothing was meant to happen until the next restart anyway...

It probably doesn't matter.